### PR TITLE
feat: Expose verticalAnchor on XAxis (and CartesianAxis)

### PR DESCRIPTION
**Pull request by petii**
## Description
This PR adds the option to manually override the _verticalAnchor_ prop in order to be able to define the rotation anchor without using a custom tick component.

## Related Issue
https://github.com/recharts/recharts/issues/3469

## Motivation and Context
The _XAxis_ component has the _angle_ prop, which lets the user add a rotation to the tick labels. The horizontal anchor point for this rotation can be set through the _textAnchor_ prop, but the vertical anchor is calculated in _CartesianAxis_ internally.
```typescript
  getTickVerticalAnchor() {
    const { orientation, mirror } = this.props;
    let verticalAnchor = 'end';

    switch (orientation) {
      case 'left':
      case 'right':
        verticalAnchor = 'middle';
        break;
      case 'top':
        verticalAnchor = mirror ? 'start' : 'end';
        break;
      default:
        verticalAnchor = mirror ? 'end' : 'start';
        break;
    }

    return verticalAnchor;
  }
```
Thus for the X axis, the vertical anchor can only be `start` or `end` which can result in some visual discrepancies:
code | result
--|--
`<XAxis dataKey="name" angle={90} textAnchor="start" />` | Screenshot 2023-03-21 at 11 20 03

## Screenshots (if appropriate):
code | result
--|--
`<XAxis dataKey="name" angle={90} textAnchor="start" verticalAnchor="middle" />` |  Screenshot 2023-03-21 at 11 22 27

https://user-images.githubusercontent.com/2715418/226579948-9270987d-6358-42db-95c8-cb23fed8f056.mov

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/cartesian/CartesianAxis.tsx [hunk 1]
```diff
@@ -43,6 +43,8 @@ export interface CartesianAxisProps {
   tickFormatter?: (value: any, index: number) => string;
   ticksGenerator?: (props?: CartesianAxisProps) => CartesianTickItem[];
   interval?: number | 'preserveStart' | 'preserveEnd' | 'preserveStartEnd' | 'equidistantPreserveStart';
+  angle?: number;
```
Comment (nikolasrieble) [inline src/cartesian/CartesianAxis.tsx:46]: Why add the angle here? Seems like an unrelated change.

File: storybook/stories/API/cartesian/CartesianAxis.stories.tsx [hunk 2]
```diff
@@ -77,6 +77,78 @@ export const TickPositioning = {
   },
 };
 
+export const TickAlignment = {
```
Comment (nikolasrieble) [inline storybook/stories/API/cartesian/CartesianAxis.stories.tsx:80]: Nit: Given that this PR adds a new property, it would be great to also add a description to the Argument Table - as done in the Line.stories.tsx
Not required, but would be very well appreciated :)

File: src/cartesian/CartesianAxis.tsx (hunk 1 above)
Comment (petii) [reply src/cartesian/CartesianAxis.tsx:46]: I was going off of the property being present in the documentation (https://recharts.org/en-US/api/XAxis#angle) but now that I looked at it again this is only the case for the XAxis component.

Also I do wonder whether it should be 🤔 Because it might be a bit misleading to expose properties that are limited to affecting the ticklabels on the main axis components.

**Comment by ckifer** [comment]
would love to get this merged at some point if we can

**Comment by ckifer** [comment]
Would love to get this change in but this PR is stale - please re-open if we want to continue here
